examples/molecular_dynamics/example_molecular_dynamics.cpp

```diff
@@ -0,0 +1,178 @@
+  index.query(execution_space, Neighbors<MemorySpace>{particles, r}, indices,
+              offsets);
+
+  Kokkos::View<float * [3], MemorySpace> forces("forces", n);
+  Kokkos::parallel_for(
+      "compute_forces",
+      Kokkos::RangePolicy<ExecutionSpace>(execution_space, 0, n),
+      KOKKOS_LAMBDA(int i) {
+        auto const x_i = particles(i)[0];
+        auto const y_i = particles(i)[1];
+        auto const z_i = particles(i)[2];
+        float fxi = 0.f;
+        float fyi = 0.f;
+        float fzi = 0.f;
+        for (int j = offsets(i); j < offsets(i + 1); ++j)
+        {
+          auto const dx = x_i - particles(indices(j))[0];
+          auto const dy = y_i - particles(indices(j))[1];
+          auto const dz = z_i - particles(indices(j))[2];
+          auto const rsq = dx * dx + dy * dy + dz * dz;
+          auto const cutoff_sq_ij = KokkosExt::ArithmeticTraits::infinity<
+              std::remove_cv_t<decltype(rsq)>>::value; // FIXME
+          if (rsq < cutoff_sq_ij)
+          {
+            auto const r2inv = static_cast<decltype(rsq)>(1) / rsq;
+            auto const r6inv = r2inv * r2inv * r2inv;
+            auto const lj1_ij = 1.f; // FIXME
+            auto const lj2_ij = 1.f; // FIXME
+            auto const fij = (r6inv * (lj1_ij * r6inv - lj2_ij)) * r2inv;
+            fxi += dx * fij;
+            fyi += dy * fij;
+            fzi += dz * fij;
+          }
+        }
+        forces(i, 0) += fxi;
+        forces(i, 1) += fyi;
+        forces(i, 2) += fzi;
+      });
+
+  float potential_energy;
+  Kokkos::parallel_reduce(
+      "compute_potential_energy",
+      Kokkos::RangePolicy<ExecutionSpace>(execution_space, 0, n),
+      KOKKOS_LAMBDA(int i, float &local_energy) {
+        auto const x_i = particles(i)[0];
+        auto const y_i = particles(i)[1];
+        auto const z_i = particles(i)[2];
+        for (int j = offsets(i); j < offsets(i + 1); ++j)
+        {
+          auto const dx = x_i - particles(indices(j))[0];
+          auto const dy = y_i - particles(indices(j))[1];
+          auto const dz = z_i - particles(indices(j))[2];
+          auto const rsq = dx * dx + dy * dy + dz * dz;
+          auto const cutoff_sq_ij = KokkosExt::ArithmeticTraits::infinity<
+              std::remove_cv_t<decltype(rsq)>>::value; // FIXME
+          if (rsq < cutoff_sq_ij)
+          {
+            auto const r2inv = static_cast<decltype(rsq)>(1) / rsq;
+            auto const r6inv = r2inv * r2inv * r2inv;
+            auto const lj1_ij = 1.f; // FIXME
+            auto const lj2_ij = 1.f; // FIXME
+            local_energy = .5f * r6inv * (.5f * lj1_ij * r6inv - lj2_ij) / 6.f;
+          }
+        }
+      },
+      potential_energy);
+
+  Kokkos::parallel_for(
+      "update_particles_position_and_velocity",
+      Kokkos::RangePolicy<ExecutionSpace>(execution_space, 0, n),
+      KOKKOS_LAMBDA(int i) {
+        auto const dt = 1.f;     // FIXME
+        auto const mass_i = 1.f; // FIXME
```

Comment:
Here and other places please expand on FIXME. What exactly should be done? Do you expect to fix the FIXMEs prior to merging?

Reply:
That's a question for @streeve 
I assume this is fine together with the other parameters

Reply:
Mass of 1 is fine